### PR TITLE
Test-MtCisPasswordExpiry: Only Check domains with isVerified: true to avoid false positives

### LINKED ISSUE #1714: 🪲 CIS.M365.1.3.1 Test-MtCisPasswordExpiry failing though setting activated
### Pre-submission checklist

- [x] I can reproduce this issue using the latest released version of Maester.
- [x] I have searched the existing issues and did not find a similar report.

### Describe the bug

The test CIS.M365.1.3.1 Test-MtCisPasswordExpiry is failing and printing a result per domain although the setting is set as explained in the test.
It seems to be related to whether domains are fully set up as authorative for MS365 or not. The two domains failing aren't (for correct business reasons).

Whether the test is testing something more detailed (per domian) --> mitigation explanation should match this _or_ the test needs to match the possible settings.

### Steps to reproduce

Have a domain not reporting as healthy in Settings > Domains:
Image

Test results Maester:
Image

Settings > Security & Privacy > Password expiration policy is enabled:
Image

### Expected behavior

a) Test needs really to test the setting behaviour (global, not per domain) as pointed out in test explanation text and setting possibility _or_

b) Test is fetching some per domain result and explanation should also point out how to set per domain (if possible at all)

### Debug output

```text

```

### Module version

```text
2.0.0
```

### Environment data

```text
PowerShell Version 7.4.6
Edition Core
Platform Unix
```

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by blindzero**
## 📑 Description

`Test-MtCisPasswordExpiry` was checking on `PasswordValidityPeriodInDays` and `isManaged: true` for domains.
With unverified domains this led to false positive errors, as unverified domains have `PasswordValidityPeriodInDays: false`, although the setting is globally applied in Microsoft 365 Admin Portal UI.

My assumption is that unverified domains can't be properly used in Microsoft 365 anyways and therefor should be excluded from this test. As soon as a domain is verified, the proper `PasswordValidityPeriodInDays` value from tenant configuration is applied (checked this with a new domain on our end).

## Changes

`$result` filter was enriched with `-and ($isVerified -eq $true)`

Closes #1714

## ✅ Checks
- [X] My pull request adheres to the code style of this project.
- [ ] My code requires changes to the documentation.
- [ ] I have updated the documentation as required.
- [X] The build and unit tests pass after running `/powershell/tests/pester.ps1` locally.

### CODE REVIEW

**Review by Mynster9361 — APPROVED** [review]
Although this is not listed in the CIS documentation for this test i believe this addition is needed to avoid false positives.

LGTM

**Comment by SamErde** [comment]
Thanks, @blindzero! Great catch!
Thanks for adding a review, @Mynster9361!

**Review by SamErde — DISMISSED** [review]

**Comment by SamErde** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/maester365/maester/pull/1757#pullrequestreview-4262213341)

File: powershell/public/cis/Test-MtCisPasswordExpiry.ps1 [hunk 1]
```diff
@@ -29,7 +29,7 @@
         $domains = Invoke-MtGraphRequest -RelativeUri 'domains'
 
         Write-Verbose 'Get domains where passwords are set to expire'
-        $result = $domains | Where-Object { ($_.PasswordValidityPeriodInDays -ne '2147483647') -and ($_.authenticationType -eq "Managed") }
+        $result = $domains | Where-Object { ($_.PasswordValidityPeriodInDays -ne '2147483647') -and ($_.authenticationType -eq "Managed") -and ($_.isVerified -eq $true) }
```
[the remark this answers is not in the document]
Comment (blindzero) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1]: Hi @SamErde 
hm, what do you think...I tend to say that Copilot might be technically right, but for me this looks somehow like overdoing it? What is your project policy here? Doing it fully technical right and blowing up tests completely or KIS?

[the remark this answers is not in the document]
Comment (SamErde) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1]: I like to err on the side of caution and this update looks simple enough to bring in. It covers edge cases and unknowns, but the code is still very readable.

**Comment by blindzero** [comment]
@SamErde added safety net for string value of passwordExpiry value. Ready for review. Thx.

File: powershell/public/cis/Test-MtCisPasswordExpiry.ps1 [hunk 2]
```diff
@@ -28,8 +28,29 @@
...
-        Write-Verbose 'Get domains where passwords are set to expire'
-        $result = $domains | Where-Object { ($_.PasswordValidityPeriodInDays -ne '2147483647') -and ($_.authenticationType -eq "Managed") }
+        Write-Verbose 'Get verified and managed domains where passwords are set to expire'
+
+        $noPasswordExpiryPeriodInDays = [int]::MaxValue
+
+        $result = $domains | Where-Object {
+            # Filter out domains that are not 'managed' or not verified, as password policies do not apply to them
+            if (($_.authenticationType -ne "Managed") -or ($_.isVerified -ne $true)) {
+                return $false
```
[the remark this answers is not in the document]
Comment (SamErde) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1]: @copilot apply changes based on [this feedback](https://github.com/maester365/maester/pull/1757#discussion_r3260161961)

[the remark this answers is not in the document]
Comment (SamErde) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1]: FYI, @blindzero.

[the remark this answers is not in the document]
Comment (blindzero) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1]: @SamErde good one...thanks. implemented skip result logic. All tests ✅

**Review by SamErde — DISMISSED** [review]
LGTM

File: powershell/public/cis/Test-MtCisPasswordExpiry.ps1 [hunk 3]
```diff
@@ -28,8 +28,36 @@
...
-        $result = $domains | Where-Object { ($_.PasswordValidityPeriodInDays -ne '2147483647') -and ($_.authenticationType -eq "Managed") }
+        Write-Verbose 'Get verified and managed domains where passwords are set to expire'
+
+        $noPasswordExpiryPeriodInDays = [int]::MaxValue
+
+        $excludedDomains = @()
+        $applicableDomains = @()
+        foreach ($domain in $domains) {
+            # Password policy checks apply only to managed and verified domains.
+            if (($domain.authenticationType -ne "Managed") -or ($domain.isVerified -ne $true)) {
```
[the remark this answers is not in the document]
Comment (blindzero) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1:39]: fixed markdownresult output to clarify managed and verified domains

File: powershell/public/cis/Test-MtCisPasswordExpiry.ps1 [hunk 4]
```diff
@@ -28,8 +28,36 @@
         Write-Verbose 'Get domain details the password expiry period'
```
[the remark this answers is not in the document]
Comment (blindzero) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1]: fixed

File: powershell/public/cis/Test-MtCisPasswordExpiry.ps1 [hunk 5]
```diff
@@ -43,7 +71,9 @@
         $resultMd += "| --- | --- |`n"
         foreach ($item in $domains) {
             $itemResult = '❌ Fail'
-            if ($item.id -notin $result.id) {
+            if ($item.id -in $excludedDomains.id) {
+                $itemResult = '⏭️ Skip'
```
[the remark this answers is not in the document]
Comment (blindzero) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1:75]: @SamErde honestly I don't get the current used symbol from UX perspective. Would stay with mine "fast-forward-icon" ;-)

File: powershell/public/cis/Test-MtCisPasswordExpiry.ps1 [hunk 6]
```diff
@@ -43,7 +71,9 @@
         $resultMd += "| --- | --- |`n"
```
[the remark this answers is not in the document]
Comment (blindzero) [reply powershell/public/cis/Test-MtCisPasswordExpiry.ps1]: Fixed table headers

**Comment by blindzero** [comment]
@SamErde I hope that copilot found everything it could find in this little change ;-)

**Comment by SamErde** [comment]
> @SamErde I hope that copilot found everything it could find in this little change ;-)

It often finds things that I miss or don't think to check for. A large context with perfect memory is a nice thing to have! 😅

**Review by SamErde — APPROVED** [review]
